lib/api/objectPutRetention.js

```diff
@@ -79,14 +79,14 @@ function objectPutRetention(authInfo, request, log, callback) {
                 (err, retentionInfo) => next(err, bucket, retentionInfo, objectMD));
         },
         (bucket, retentionInfo, objectMD, next) => {
-            // eslint-disable-next-line no-param-reassign
-            objectMD.retentionInfo = retentionInfo;
+            /* eslint-disable no-param-reassign */
```

Comment:
To be clean, you should re-enable this warning down in the code as soon as you don't need it for your specific purpose. (whether the linter warning is useful or not is another story :wink:)